scala/private/common.bzl

```diff
@@ -172,22 +172,25 @@ def create_java_provider(scalaattr, transitive_compile_time_jars):
             output_jar = jar,
             compile_jar = jar
         )
-        for jar in transitive_compile_time_jars
+        #TODO: Avoid flattening the depset here.
+        for jar in transitive_compile_time_jars.to_list()
     ]
 
     compile_providers = [
         JavaInfo(
             output_jar = jar,
             compile_jar = jar,
             deps = compile_time_deps)
-        for jar in scalaattr.compile_jars
+        #TODO: Avoid flattening the depset here.
+        for jar in scalaattr.compile_jars.to_list()
     ]
     runtime_providers = [
         JavaInfo(
             output_jar = jar,
             compile_jar = jar
         )
-        for jar in scalaattr.transitive_runtime_jars
+        #TODO: Avoid flattening the depset here.
+        for jar in scalaattr.transitive_runtime_jars.to_list()
     ]
 
-    return java_common.merge(compile_providers, runtime_providers)
+    return java_common.merge(compile_providers + runtime_providers)
```

Comment:
Rather than using + here can we make a list like all_providers and extend into it. I think this should be better performance since skylark does not have sublinear list concat AFAIK.

Reply:
Sorry I missed this comment before. I replaced all `+` usages for `java_common.merge` with list extensions.